### PR TITLE
add python3 compatibility

### PR DESCRIPTION
**Pull request by anneFly**

### CODE REVIEW

File: django_pgviews/view.py [hunk 1]
```diff
@@ -130,47 +133,45 @@ def clear_view(connection, view_name, materialized=False):
...
-                elif isinstance(field_name, basestring):
-                    match = FIELD_SPEC_RE.match(field_name)
-                    if not match:
-                        raise TypeError("Unrecognized field specifier: %r" %
-                                        field_name)
-                    deferred_projections.append(match.groups())
-                else:
+class ViewMeta(models.base.ModelBase):
+    def __new__(metacls, name, bases, attrs):
+        '''Deal with all of the meta attributes, removing any Django does not want
```
Comment (codingjoe) [inline django_pgviews/view.py]: You should use triple double quotes

**Comment by codingjoe** [comment]
Test config for py3 is missing

**Comment by anneFly** [comment]
@codingjoe thanks but this is not ready for review yet, I'm still gonna clean up, that's why it's assigned to me...

**Comment by codingjoe** [comment]
Sorry, just popped up on my notifications and I stared looking into it right away.
BTW, you might want to push this up stream.

**Comment by anneFly** [comment]
@codingjoe now you can :)
What did you mean by 
> Test config for py3 is missing

Are you talking about the `circle.yml` file?

**Comment by codingjoe** [comment]
@anneFly, yes I'm talking about the circle ci file.

**Comment by anneFly** [comment]
@codingjoe I added the version to the circle.yml, but I'm not sure if that's the way to go. The package is supposed to be compatible in both versions, so it should theoretically be tested in both versions, but it seems like it's not too easy to do this.
Anyway, I'm planning to make a PR to the original Repo. We'll see if it will be accepted ot declined...

**Comment by syphar** [comment]
@anneFly another problem found: 

- take a view
- rename a column (that means `force` = drop and create is needed) 
- other views depend on it

File: circle.yml [hunk 2]
```diff
@@ -1,3 +1,7 @@
+machine:
+  python:
+    version: 3.5.0
```
Comment (codingjoe) [inline circle.yml]: please add `2.7,3.4,3.5`

File: django_pgviews/view.py [hunk 3]
```diff
@@ -130,47 +141,45 @@ def clear_view(connection, view_name, materialized=False):
...
-                elif isinstance(field_name, basestring):
-                    match = FIELD_SPEC_RE.match(field_name)
-                    if not match:
-                        raise TypeError("Unrecognized field specifier: %r" %
-                                        field_name)
-                    deferred_projections.append(match.groups())
-                else:
+class ViewMeta(models.base.ModelBase):
+    def __new__(metacls, name, bases, attrs):
+        """Deal with all of the meta attributes, removing any Django does not want
```
Comment (codingjoe) [inline django_pgviews/view.py]: Single line, with a dot.

File: django_pgviews/view.py [hunk 4]
```diff
@@ -130,47 +141,45 @@ def clear_view(connection, view_name, materialized=False):
...
+            _DEFERRED_PROJECTIONS[model_spec][view_cls].append(field_name)
+            _realise_projections(app_label, model_name)
 
-            return view_cls
+        return view_cls
 
-    __metaclass__ = ViewMeta
+
+class View(six.with_metaclass(ViewMeta, models.Model)):
+    """Helper for exposing Postgres views as Django models.
```
Comment (codingjoe) [inline django_pgviews/view.py]: Single line

**Comment by codingjoe** [comment]
Looks good, besides my comments. Please open upstream PR asap so we can get the maintainers feedback.

**Comment by codingjoe** [comment]
I'm actually gonna close this one in favor of an upstream PR.

**Comment by syphar** [comment]
@codingjoe we were talking about this, that we can go live with this version when it's on our version, then go for the upstream PR. 

at least as I remember it

**Comment by codingjoe** [comment]
I don't really think its a good idea to reference any commits that are not in a long term supported branch.
@anneFly You'll have to create a new branch based on this one and apply your changes there since we apparently have this already in production.
@syphar I really don't get why we have unreviewed code on production, nor why we didn't push this upstream in the beginning since we agreed to recontribute all OS changes.

**Comment by syphar** [comment]
- it wasn’t unreviewed
- @anneFly will finish it up




--  
Denis Cornehl
Vice President Software Engineering  
  
M +49 151 25251450
denis.cornehl@thermondo.de (mailto:denis.cornehl@thermondo.de)
°Thermondo
- Der Heizungsbauer.
We are hiring: www.thermondo.de/jobs/ (http://www.thermondo.de/jobs/)
www.thermondo.de (http://www.thermondo.de/) | Thermondo GmbH, Brunnenstr. 153, 10115 Berlin
Eingetragen in Berlin unter: HRB 129321 I Steuernummer: 30/557/33244
Geschäftsführer: Florian Tetzlaff, Philipp A. Pausder | USt-IdNr.: DE273951924
Facebook (https://www.facebook.com/thermondo) | Google+ (https://plus.google.com/+ThermondoDe/posts) | Twitter (https://twitter.com/thermondo) | Youtube (https://www.youtube.com/channel/UCaM56yLLGe4y4qohjjKFAYA)








Am Dienstag, 5. April 2016 um 12:28 schrieb Johannes Hoppe:

File: circle.yml (hunk 2 above)
Comment (anneFly) [inline circle.yml]: @codingjoe I cannot find anything in the circle ci documentation that this is possible. I will try to do a different workaround for this.

**Pull request closed without merging**